crates/oxc_parser/src/ts/types.rs

```diff
@@ -353,7 +353,9 @@ impl<'a> ParserImpl<'a> {
             // // falls through
             // case SyntaxKind.FunctionKeyword:
             // return parseJSDocFunctionType();
-            Kind::Question => self.parse_js_doc_unknown_or_nullable_type(),
+            Kind::Question if self.ctx.contains(Context::AllowQuestionMarkAfterType) => {
```

Comment:
Could we parse the question mark and report an error instead of panicking? It would be nice to display a descriptive error message. [Example](https://www.typescriptlang.org/play/?#code/C4TwDgpgBAKlC8UDaAPAXFAdgVwLYCMIAnAfgF0BuIA) from TypeScript:

> A labeled tuple element is declared as optional with a question mark after the name and before the colon, rather than after the type. (5086)